### PR TITLE
docs: use test: all rather than editing .ci.yaml

### PR DESCRIPTION
**Pull request by cbracken**
Rather than editing `.ci.yaml` to enable the `linux_web_engine` shard, instead suggest tagging with the `test: all` label, which avoids the risk of accidentally committing an unwanted change to `.ci.yaml` and having to revert the change before landing.

This is also more future-proof against further `runIf` additions to `.ci.yaml`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: docs/Testing-presubmit-Engine-PRs-with-the-Flutter-framework.md [hunk 1]
```diff
@@ -4,16 +4,26 @@ This documentation describes how to run flutter/flutter presubmit checks on flut
...
 3. Determine the commit hash for your flutter/engine PR.
 4. Create and upload a flutter/flutter PR, (OR run tests locally).
 5. Wait for flutter/flutter presubmits/tests to run ☕.
-6. Edit `.ci.yaml` to uncomment the `runIf` block you commented out above.
 
-## 1. Edit .ci.yaml
+## 1. Create your engine pull request with the `test: all` label.
 
-Edit [`.ci.yaml`](https://github.com/flutter/engine/blob/main/.ci.yaml) to comment out the `runIf:` block in `linux_web_engine`. This will ensure Flutter Web artifacts build, otherwise most framework tests will fail during precache.
+When creating your PR, add the `test: all` label *before* submitting the PR to
```
Comment (jmagman) [inline docs/Testing-presubmit-Engine-PRs-with-the-Flutter-framework.md:15]: What does "before submitting the PR to presubmit" mean?

File: docs/Testing-presubmit-Engine-PRs-with-the-Flutter-framework.md [hunk 2]
```diff
@@ -4,16 +4,26 @@ This documentation describes how to run flutter/flutter presubmit checks on flut
...
+presubmit. This will ensure that all builds required for framework testing are
+triggered.
+
+If you sent out your PR without adding the `test: all` label, add it, then
+re-push your branch to re-trigger presubmits.
+
+By default, not all builds and tests are run in engine presubmits. When our CI
```
Comment (jmagman) [inline docs/Testing-presubmit-Engine-PRs-with-the-Flutter-framework.md:22]: Optionally link to ci/Engine-pre-submits-and-post-submits.md#running-post-submits-eagerly


Relative markdown links https://github.blog/news-insights/product-news/relative-links-in-markup-files/

File: docs/Testing-presubmit-Engine-PRs-with-the-Flutter-framework.md (hunk 1 above)
Comment (cbracken) [reply docs/Testing-presubmit-Engine-PRs-with-the-Flutter-framework.md:15]: That time when you're still creating the PR but before you hit the big green create button (or select `submit` in the `gh` tool) that triggers the presubmits.

There's got to be a better name for that state, but I struggled for a moment before coming up with this phrasing, which admittedly is pretty awkward.

File: docs/Testing-presubmit-Engine-PRs-with-the-Flutter-framework.md (hunk 2 above)
Comment (cbracken) [reply docs/Testing-presubmit-Engine-PRs-with-the-Flutter-framework.md]: Good call -- done.

File: docs/Testing-presubmit-Engine-PRs-with-the-Flutter-framework.md (hunk 1 above)
Comment (cbracken) [reply docs/Testing-presubmit-Engine-PRs-with-the-Flutter-framework.md:15]: Reworded as "before you submit to CI" which is maybe a bit clearer?

**Review by jmagman — APPROVED** [review]

**Pull request merged**